File: algorithm/java/src/main/java/com/liyongquan/weeklycontest/wc304/MinimumOperations.java
package com.liyongquan.weeklycontest.wc304;

import java.util.Arrays;

public class MinimumOperations {
    public int minimumOperations(int[] nums) {
        Arrays.sort(nums);
        int res = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                int tmp = nums[i];
                for (int j = i; j < nums.length; j++) {
                    nums[j] -= tmp;
                }
                res++;
            }
        }
        return res;
    }
}

File: algorithm/golang/weekcontest/wc429/minimumOperations.go
package wc429

// 思路反过来，从右往左添加元素，看至少需要增加到哪里
func minimumOperations(nums []int) int {
	mp := make(map[int]int)
	n := len(nums)
	p := 0
	for i := n - 1; i >= 0; i-- {
		mp[nums[i]]++
		if mp[nums[i]] > 1 {
			p = i + 1
			break
		}
	}
	// [p,n)为最小满足条件的
	return (p + 2) / 3
}
